### ISSUE TITLE
Controlling the colors

### ISSUE THREAD

**Issue by oztalha** [geopandas/geopandas#420]
I'm trying to match the colors of two plots I create with gpd.plot() and plot() respectively.
The problem is that geopandas change the colors according to the length of the series.
I solved the problem by setting `cmap=plt.get_cmap('Set1')` but I think we better change the default behavior.

1. It appears to me that this function causes the issue:
```
def gencolor(N, colormap='Set1'):
    from matplotlib import cm
    # don't use more than 9 discrete colors
    n_colors = min(N, 9)
    cmap = cm.get_cmap(colormap, n_colors)
    colors = cmap(range(n_colors))
    for i in xrange(N):
        yield colors[i % n_colors]
```
Instead of `min(N,9)`, I think we should simply have `n_colors = 9`.
Pandas does not change the color scheme:
```
#in Pandas second color is blue
pd.DataFrame(pd.np.random.randn(2, 2)).plot()

#in GeoPandas second color is grey (I'd expect blue)
gpd.GeoSeries([Point(0,0),Point(0,1)]).plot()
```

2. It appears that Geopandas does not care about the plot style set in matplotlib. When I `plt.style.use('ggplot')` and then plot in geopandas, colors do not change (that is not the case for pandas). 

3. I tried setting colors as an iterable of hex (or rgba) while plotting, but neither worked for geopandas (though both work for pandas):
```
df = pd.DataFrame(np.random.randn(5, 5))
custom_colors = ['#FF0000', '#0000FF', '#FFFF00', '#000000', '#FFFFFF']
df.plot(color=custom_colors)
```
Setting a single color like `gpd.GeoSeries([...]).plot(color='#FF0000')` works, but passing multiple colors as a list gives `could not convert string to float: '#FF0000'` value error.

**Comment by ResidentMario** [comment]
xref #405

**Comment by samsammurphy** [comment]
So is there now a simple solution to this?

**Comment by dfolch** [comment]
I think this might be the same issue. I am trying to keep a single user defined color ramp for different attributes on the same geography. While I get the same number of bins each time, the color associated with each bin depends on the range of values.

geopandas version 0.8.1
```python
# Get some polygons
data = gpd.read_file('https://www2.census.gov/geo/tiger/TIGER2018/STATE/tl_2018_us_state.zip')
data = data.loc[~data.NAME.isin(['Alaska','Hawaii','Puerto Rico','United States Virgin Islands','Commonwealth of the Northern Mariana Islands','Guam','American Samoa']),]

# Create some fake data with different ranges
data['low_vals'] = np.random.uniform(0, 0.3, data.shape[0])
data['mid_vals'] = np.random.uniform(0.3, 0.7, data.shape[0])
data['high_vals'] = np.random.uniform(0.7, 1.0, data.shape[0])

# Create a single set of bins
bins = np.arange(1,11)/10 # set bins to 0.10 intervals
```

```python
data.plot('low_vals', scheme='UserDefined', classification_kwds={'bins':bins}, legend=True)
```
![low](https://user-images.githubusercontent.com/1322332/106776694-8c503800-6601-11eb-822d-5a889a03fec1.png)

```python
data.plot('mid_vals', scheme='UserDefined', classification_kwds={'bins':bins}, legend=True)
```
![mid](https://user-images.githubusercontent.com/1322332/106776719-91ad8280-6601-11eb-8740-7afb100acd11.png)

```python
data.plot('high_vals', scheme='UserDefined', classification_kwds={'bins':bins}, legend=True)
```
![high](https://user-images.githubusercontent.com/1322332/106776737-95410980-6601-11eb-98a7-136d28cd84d2.png)

**Comment by dfolch** [comment]
Here is a workaround using the same example as above. (cross-reference https://github.com/pysal/mapclassify/issues/109) 
```python
# Setup
import matplotlib as mpl
cmap = mpl.cm.viridis  # choose a color map
norm = mpl.colors.BoundaryNorm(np.arange(0,11)/10, cmap.N)  # set bins to 0.10 intervals
```
```python
data.plot('low_vals', cmap=cmap, norm=norm, legend=True)
```
![low_fixed](https://user-images.githubusercontent.com/1322332/111009273-6cbed480-8350-11eb-9c99-700e0119b4b6.png)
```python
data.plot('mid_vals', cmap=cmap, norm=norm, legend=True)
```
![mid_fixed](https://user-images.githubusercontent.com/1322332/111009302-7d6f4a80-8350-11eb-9981-9e7732e2422b.png)

```python
data.plot('high_vals', cmap=cmap, norm=norm, legend=True)
```
![high_fixed](https://user-images.githubusercontent.com/1322332/111009310-82cc9500-8350-11eb-940b-ec4e2bc83cb0.png)

**Comment by jorisvandenbossche** [comment]
@martinfleis your PR fixed the last comments for the bug when using a user defined classification scheme (where you can have outer bins not present in your data). But does this also close the general issue / original report?

**Comment by martinfleis** [comment]
This snippet from above works as I would expect, so no issue there.

```
#in Pandas second color is blue
pd.DataFrame(pd.np.random.randn(2, 2)).plot()

#in GeoPandas second color is grey (I'd expect blue)
gpd.GeoSeries([Point(0,0),Point(0,1)]).plot()
```

> It appears that Geopandas does not care about the plot style set in matplotlib. When I plt.style.use('ggplot') and then plot in geopandas, colors do not change (that is not the case for pandas).

Style is applied correctly.

> I tried setting colors as an iterable of hex (or rgba) while plotting, but neither worked for geopandas (though both work for pandas)

Custom array of colours works.

I don't think any of these issues were solved by #2019 but all are fixed now by other PRs that happened in the meantime.
